core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -7326,6 +7343,158 @@ private void handleRenameGroupFailure(String errorCode, String errorMessage, Str
         }
     }
 
+    /**
+     * This method is responsible for calling relevant listener methods before updating user list of a group.
+     *
+     * @param groupId        ID of the group.
+     * @param deletedUserIds List of users to be removed from the group.
+     * @param newUserIds     List of users to be added to the group.
+     * @throws UserStoreException Exception that will be thrown by relevant listeners.
+     */
+    private boolean handlePreUpdateUserListOfGroup(String groupId, List<String> deletedUserIds,
+                                                   List<String> newUserIds) throws UserStoreException {
+
+        try {
+            for (GroupOperationEventListener listener : UMListenerServiceComponent
+                    .getGroupOperationEventListeners()) {
+                if (listener instanceof AbstractGroupOperationEventListener) {
+                    AbstractGroupOperationEventListener newListener = (AbstractGroupOperationEventListener) listener;
+                    if (!newListener.preUpdateUserListOfGroup(groupId, deletedUserIds, newUserIds,
+                            this)) {
```

Comment:
Can we make this into a single line?

Reply:
will be fixed with https://github.com/wso2/carbon-kernel/pull/3852

Reply:
Ack